src/main/java/bio/terra/service/dataset/flight/LockDatasetStep.java

```diff
@@ -15,24 +14,23 @@
 public class LockDatasetStep implements Step {
 
     private DatasetDao datasetDao;
-    private DatasetRequestModel datasetRequest;
+    private String datasetName;
 
     private static Logger logger = LoggerFactory.getLogger(LockDatasetStep.class);
 
-    public LockDatasetStep(DatasetDao datasetDao, DatasetRequestModel datasetRequest) {
+    public LockDatasetStep(DatasetDao datasetDao, String datasetName) {
         this.datasetDao = datasetDao;
-        this.datasetRequest = datasetRequest;
+        this.datasetName = datasetName;
     }
 
     @Override
     public StepResult doStep(FlightContext context) {
         try {
-            String datasetName = datasetRequest.getName();
-            datasetDao.lock(datasetName, context.getFlightId());
-
             FlightMap workingMap = context.getWorkingMap();
```

Comment:
It looks to me like you don't need to put DATASET_NAME in the workingMap. You could just pass the name into the unlock as you do in the lock. Is there a reason for the asymmetry?

Reply:
Yes, good point. Made the change for dataset and snapshot metadata creation steps.